### PR TITLE
Avoid fixed-point overflow in inlineCurrentTimeMaxPrecision

### PR DESCRIPTION
**Pull request by fjeremic**
inlineCurrentTimeMaxPrecision uses a signed subtract which may raise a
hardware trap if the fixed-point overflow exception flag is set in the
PSW. The result of STCKF is however a true 64-bit unsigned value and as
such we should be using logical subtracts.

Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
@joransui could I please get a review?

File: runtime/tr.source/trj9/z/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -9778,154 +9778,127 @@ static TR::Register * inlineSinglePrecisionSQRT(TR::Node *node, TR::CodeGenerato
...
+      int32_t todJanuary1970High = 0x7D91048B;
+      int32_t todJanuary1970Low = 0xCA000000;
 
-      // tmpReg2 is not actually used
-      //
-      generateS390ImmOp(cg, TR::InstOpCode::SL, node, tmpReg2, targetLowRegister, todJan1970L, NULL, litBase);
-      generateS390ImmOp(cg, TR::InstOpCode::SLB, node, tmpReg2, targetHighRegister, todJan1970H, NULL, litBase);
+      // tempRegister2 is not actually used
+      generateS390ImmOp(cg, TR::InstOpCode::SL, node, tempRegister2, targetRegister->getLowOrder(), todJanuary1970Low, NULL, literalBaseRegister);
+      generateS390ImmOp(cg, TR::InstOpCode::SLB, node, tempRegister2, targetRegister->getHighOrder(), todJanuary1970High, NULL, literalBaseRegister);
```
Comment (fjeremic) [inline runtime/tr.source/trj9/z/codegen/J9TreeEvaluator.cpp]: I had concerns that on 31-bit we were not actually subtracting 1970 but this is not the case. The `generateS390ImmOp` API aliases the source with the target register for `SL` and `SLB`. Everything should work as expected.

**Comment by fjeremic** [comment]
@joransiu could I please get a review?

**Comment by joransiu** [comment]
LGTM

**Comment by joransiu** [comment]
jenkins sanity zLinux

**Comment by fjeremic** [comment]
Any other concerns @joransiu?

**Pull request merged**